connection/connection.go

```diff
@@ -0,0 +1,69 @@
+package connection
+
+import (
+	"context"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-csi/csi-lib-utils/protosanitizer"
+	"google.golang.org/grpc"
+)
+
+const (
+	// Interval of logging connection errors
+	connectionLoggingInterval = 10 * time.Second
+)
+
+// Connect opens insecure gRPC connection to a CSI driver. Address must have either '<protocol>://' prefix, or be
+// a path to a socket file. The function tries to connect indefinitely every second until it connects.
+// The function automatically adds interceptor for gRPC message logging.
+func Connect(address string, dialOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
+	dialOptions = append(dialOptions,
+		grpc.WithInsecure(),                   // Don't use TLS, it's usually local Unix domain socket in a container.
+		grpc.WithBackoffMaxDelay(time.Second), // Retry every second after failure.
+		grpc.WithBlock(),                      // Block until connection succeeds.
+		grpc.WithUnaryInterceptor(LogGRPC),    // Log all messages.
+	)
+	if strings.HasPrefix(address, "/") {
+		// It looks like filesystem path.
+		dialOptions = append(dialOptions, grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+			return net.DialTimeout("unix", addr, timeout)
+		}))
+	}
+	glog.Infof("Connecting to %s", address)
+
+	// Wait until the connection is ready.
+	var conn *grpc.ClientConn
+	var err error
+	ready := make(chan bool)
+	go func() {
+		conn, err = grpc.Dial(address, dialOptions...)
+		close(ready)
+	}()
+
+	// Log error every connectionLoggingInterval
+	ticker := time.NewTicker(connectionLoggingInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			glog.Warningf("Still connecting to %s", address)
```

Comment:
`glog` has been replaced in Kubernetes. We should do the same in kubernetes-csi. Can you replace with `k8s.io/klog` here and in the rest of the new code?

Reply:
We should move all repos to klog in one (short) burst. Is it the right time?

Reply:
I agree, but we should run it through CSI WG meeting. And add common logging initialization function to this library (as separate PR), so all the sidecar containers have the same cmdline arguments for logging.